Assets/LeapMotionModules/UI/Interaction/InteractionHand.cs

```diff
@@ -0,0 +1,780 @@
+      body.mass = PER_BONE_MASS_MULTIPLIER * massScale * brushBone._lastObjectTouchedMass;
+
+      //If these conditions are met, stop using brush hands to contact objects and switch to "Soft Contact"
+      if (!_softContactEnabled && targetingError >= DISLOCATION_FRACTION
+          && _hand.PalmVelocity.Magnitude < 1.5f && boneArrayIndex != NUM_FINGERS * BONES_PER_FINGER) {
+        EnableSoftContact();
+        return;
+      }
+
+      // Add a deadzone to avoid vibration.
+      Vector3 delta = bone.Center.ToVector3() - body.position;
+      float deltaLen = delta.magnitude;
+      if (deltaLen <= deadzone) {
+        body.velocity = Vector3.zero;
+        brushBone.lastTarget = body.position;
+      }
+      else {
+        delta *= (deltaLen - deadzone) / deltaLen;
+        brushBone.lastTarget = body.position + delta;
+        delta /= Time.fixedDeltaTime;
+        body.velocity = (delta / delta.magnitude) * Mathf.Clamp(delta.magnitude, 0f, 100f);
+      }
+    }
+
+    private bool _softContactEnabled = false;
+    private bool _disableSoftContactEnqueued = false;
+    private IEnumerator _delayedDisableSoftContactCoroutine;
+    private Collider[] _tempColliderArray = new Collider[2];
+    private Vector3[] _previousBoneCenters = new Vector3[20];
+    private float _softContactBoneRadius = 0.015f;
+
+    // Vars removed because Interaction Managers are currently non-optional.
+    //private List<PhysicsUtility.SoftContact> softContacts = new List<PhysicsUtility.SoftContact>(40);
+    //private Dictionary<Rigidbody, PhysicsUtility.Velocities> originalVelocities = new Dictionary<Rigidbody, PhysicsUtility.Velocities>();
+
+    private void FixedUpdateSoftContact() {
+      if (_hand == null) return;
+
+      if (_softContactEnabled) {
+
+        // Generate contacts.
+        bool softlyContacting = false;
+        for (int fingerIndex = 0; fingerIndex < 5; fingerIndex++) {
+          for (int jointIndex = 0; jointIndex < 4; jointIndex++) {
+            Bone bone = _hand.Fingers[fingerIndex].Bone((Bone.BoneType)(jointIndex));
+            int boneArrayIndex = fingerIndex * 4 + jointIndex;
+            Vector3 boneCenter = bone.Center.ToVector3();
+
+            // Generate and fill softContacts with SoftContacts that are intersecting a sphere
+            // at boneCenter, with radius softContactBoneRadius.
+            bool sphereIntersecting;
+            sphereIntersecting = PhysicsUtility.GenerateSphereContacts(boneCenter, _softContactBoneRadius,
+                                                                       (boneCenter - _previousBoneCenters[boneArrayIndex])
+                                                                         / Time.fixedDeltaTime,
+                                                                       1 << interactionManager.InteractionLayer,
+                                                                       ref interactionManager._softContacts,
```

Comment:
Instead of giving the manager a public interface so hands can talk back to it, how about the hands can return some results as a result of being updated, that the manager can then act on?